BIT Reminder/BIT Reminder/MVPGroups/HomeMVP/HomeViewController.swift

```diff
@@ -314,47 +278,36 @@ extension HomeViewController: UITableViewDelegate, UITableViewDataSource {
         } else {
             debugPrint("No row selected")
         }
+}
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
     }
-    
-    private func createSpinnerFooter() -> UIView {
-        let footerView = UIView(frame: CGRect(x: 0, y: 0, width: view.frame.size.width, height: 100))
-        let spinner = UIActivityIndicatorView()
-        spinner.color = .white
-        spinner.center = footerView.center
-        footerView.addSubview(spinner)
-        spinner.startAnimating()
-        
-        return footerView
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let sectionHeaderTitle = selectedFilterTitle else {
+            return nil // Hide section header if no filter is selected
+        }
+
+        guard let sectionHeader = tableView.dequeueReusableHeaderFooterView(withIdentifier: FilterSectionView.identifier) as? FilterSectionView else {
+            fatalError("section header failed")
+        }
+        sectionHeader.configure(with: sectionHeaderTitle)
+        return sectionHeader
     }
-    
-    // MARK: detecting user scroll did reach bootom- start paginating
-    
-    func scrollViewDidEndDragging(_ scrollView: UIScrollView, willDecelerate decelerate: Bool) {
-        let offsetY = scrollView.contentOffset.y
-        let contentHeight = scrollView.contentSize.height
-        
-        if offsetY > contentHeight - scrollView.frame.size.height {
-            if self.homeView.tableView.tableFooterView == nil {
-                debugPrint("Adding spinner footer")
-                self.homeView.tableView.tableFooterView = createSpinnerFooter()
-                isPaginating = true
-                presenter.itemsPerPage += 1
-                self.loadReminders()
-            }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
```

Comment:
Maybe this could be written like this one line command:

```
func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
    return selectedFilterTitle == nil ? 0 : 20
}
```